.travis.yml

```diff
@@ -5,39 +5,50 @@ android:
   components:
     - platform-tools
     - tools
-    - build-tools-22.0.0
-    - android-21
+    - build-tools-27.0.3
+    - android-25
+    - extra-google-m2repository
+    - extra-android-m2repository
+  licenses:
+   - android-sdk-preview-license-.+
+   - android-sdk-license-.+
 sudo: false
+before_cache:
+  - rm -f  $HOME/.gradle/caches/modules-2/modules-2.lock
+  - rm -fr $HOME/.gradle/caches/*/plugin-resolution/
+cache:
+  directories:
+    - $HOME/.gradle/caches/
+    - $HOME/.gradle/wrapper/
+    - $HOME/.android/build-cache
 before_install:
 - test -z "$encrypted_c726d225a9d9_key" || openssl aes-256-cbc -K $encrypted_c726d225a9d9_key -iv $encrypted_c726d225a9d9_iv
   -in debug.keystore.enc -out debug.keystore -d
 - GIT_HASH=`git log --pretty=format:'%h' -n 1`
 - GIT_BUILD=`git describe --all --always`-$GIT_HASH
+- mkdir "$ANDROID_HOME/licenses" || true
+- echo -e "\n8933bad161af4178b1185d1a37fbf41ea5269c55" > "$ANDROID_HOME/licenses/android-sdk-license"
+- echo -e "\n84831b9409646a918e30573bab4c9c91346d8abd" > "$ANDROID_HOME/licenses/android-sdk-preview-license"
+- chmod +x shell/android-studio/gradlew
 install:
 - sudo apt-get -qq update
-- sudo apt-get install ant
-- wget http://dl.google.com/android/ndk/android-ndk-r10d-linux-x86_64.bin
-- chmod +x android-ndk-r10d-linux-x86_64.bin
-- ./android-ndk-r10d-linux-x86_64.bin -y | grep -v Extracting
-- mv android-ndk-r10d android-ndk
-- rm android-ndk-r10d-linux-x86_64.bin
-- export NDK=$(pwd)/android-ndk
-- export ANDROID_NDK=$(pwd)/android-ndk
+- touch ~/.android/repositories.cfg
+- echo y | sdkmanager 'ndk-bundle'
+- export NDK=ANDROID_NDK=ANDROID_NDK_HOME=$ANDROID_HOME/ndk-bundle
 - export PATH=${PATH}:$ANDROID_HOME/tools
 - export PATH=${PATH}:$ANDROID_HOME/platform-tools
 - export PATH=${PATH}:$NDK
 - mkdir -p ~/.android
 - test -z "$encrypted_c726d225a9d9_key" || mv debug.keystore ~/.android/debug.keystore
 before_script: 
 script:
-- cd shell/android
-- android update project -p . --target "android-21"
+- cd shell/android-studio
 - export NUMBER_OF_PROCESSORS=2
-- ant debug
+- ./gradlew build
```

Comment:
Can we also add a release build here?

Reply:
Once there is one, you can set assembleRelease or assembleDebug.